src/opm/parser/eclipse/EclipseState/Schedule/Network/Balance.cpp

```diff
@@ -64,11 +65,11 @@ Balance::Balance(const Tuning& tuning, const DeckKeyword& keyword) {
 Balance::Balance(bool network_active, const Tuning& tuning)
     : calc_mode(CalcMode::TimeStepStart)
     , calc_interval(0)
-    , m_thp_tolerance( NB::THP_CONVERGENCE_LIMIT::defaultValue )
+    , m_thp_tolerance( unit::barsa * NB::THP_CONVERGENCE_LIMIT::defaultValue )
```

Comment:
Isn't this supposed to happen automatically?  Did we miss a `dimension` tag somewhere?  Another option would be to write it as
```C++
UnitSystem::newMETRIC().to_si(UnitSystem::measure::pressure, NB::THP_CONVERGENCE_LIMIT::defaultValue
```
instead.  That way we won't have to pollute more implementation files with explicit knowledge about unit conventions.

Reply:
No we need a `DeckKeyword` instance to get automatic unit conversion - the design is not better ....

Happy to change to use the `UnitSystem` AS suggested.